ts/src/index.ts

```diff
@@ -0,0 +1,60 @@
+import Discord from 'discord.js'
+import dotenv from "dotenv"
+import { PrismaClient } from "@prisma/client"
+
+dotenv.config()
+const client = new Discord.Client() 
+const prisma = new PrismaClient()
+
+async function repUpdate(userId: number, operation: string): Promise<number> {
+  const exists = await prisma.user.findUnique({where: {id: userId}}) //check if the user already exists 
+  if(!exists) {
+    operation.includes("++") ? // create the user and add rep 0 or 2 depeniing if the contents includes ++
+      await prisma.user.create({data: { id: userId, rep: 2 }}) :
+      await prisma.user.create({data: { id: userId, rep: 0 }})
+    return operation.includes("++") ? 2 : 0 // return rep for message
+  } else {
+    let user = await prisma.user.findUnique({where: {id: userId}}) // get the user
+    operation.includes("++") ?
+      await prisma.user.update({where: {id: userId}, data: { rep: Number(user!.rep + 1) }}) : // update the user
+      await prisma.user.update({where: {id: userId}, data: { rep: Number(user!.rep - 1) }})
+    return operation.includes("++") ?
+      user!.rep + 1 :
+      user!.rep - 1
+  }
+}
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`)
+});
+
+client.on('message', async (msg) => {
+  const args: any[] = msg.content.trim().split(' ') // get args
+  if (
+      msg.content.includes("++") || msg.content.includes("--") // check if the message contains -- or ++
+  ) {
+    let targetName: string = args.join(" ").replace(/(\+\+|--)\s*$/, "") // get the name
+    if(targetName.includes('@')) { // check if the name has a @ in it
+      const pingedUsed: Object = msg.mentions.users.toJSON() // get the mentioned user
```

Comment:
Possible typo pingedUsed > pingedUser
Dont you have a defined schema for this? Is this pingedUsed dynamic?

I think you should do interface PingedUser{

}

Reply:
Also i would think of calling this variable pingedUsers since this is an array of Users right?

Reply:
When I added a interface 
```ts
interface PingedUser {
  id: string,
  system?: any,
  locale?: any,
  flags: number,
  username: string,
  bot: boolean,
  discriminator: number,
  avatar?: string,
  lastMessageChannelID?: string,
  createdTimestamp: number,
  defaultAvatarURL?: string,
  tag: string,
  avatarURL?: string,
  displayAvatarURL?: string
}
```

The linter said this `TS2740: Type '{}' is missing the following properties from type 'PingedUser[]': length, pop, push, concat, and 26 more.`

Reply:
isnt toJSON() a way to tell an object how to parse it to a string? Basicly overwriting JSON.Stringify?

I think you should change that to JSON.parse(msg.mentions.users);